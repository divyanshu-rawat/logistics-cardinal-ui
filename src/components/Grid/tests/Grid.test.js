import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Cell } from 'styled-css-grid';

import ComposedGrid from '../Grid';
import theme from '../../../themes/rooster';

const renderedComponent = (withPad) =>
  shallow(
    <ComposedGrid theme={theme} withPad={withPad} columns={12}>
      <Cell width={12}>Corinthians!</Cell>
    </ComposedGrid>,
  );

describe('<Grid />', () => {
  it('should render the component with the default props', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });

  it('should render the component with `withPad` prop as true', () => {
    const tree = toJson(renderedComponent(true));
    expect(tree).toMatchSnapshot();
  });
});
