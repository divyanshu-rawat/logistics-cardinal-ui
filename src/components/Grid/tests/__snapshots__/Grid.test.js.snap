// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should render the component with \`withPad\` prop as true 1`] = `
<styled.section
  withPad={true}
>
  <Grid__Grid
    columns={12}
    theme={
      Object {
        "borderRadius": Object {
          "giga": "5px",
          "kilo": "1px",
          "mega": "4px",
        },
        "borderWidth": Object {
          "kilo": "1px",
          "mega": "2px",
        },
        "colors": Object {
          "black": "#000",
          "bodyBackground": "#292d33",
          "danger": "#a94442",
          "green100": "#c6e5cb",
          "green300": "#97e2a3",
          "green500": "#62de76",
          "green700": "#49b85b",
          "green900": "#2f903e",
          "grey100": "#96a2b0",
          "grey200": "#3f454d",
          "grey300": "#343a42",
          "grey400": "#292d33",
          "grey500": "#40464f",
          "mainBackground": "343941",
          "primary100": "#3f9bfd",
          "primary200": "#375f8a",
          "primary300": "#324e70",
          "primary400": "#96a2b0",
          "primary500": "#484e58",
          "red100": "#ffe6e1",
          "red300": "#ffaf9f",
          "red500": "#ff7559",
          "red700": "#d55a41",
          "red900": "#a94442",
          "success": "#49b85b",
          "textColor": "#fff",
          "warning": "#d4a546",
          "white": "#fff",
          "yellow100": "#f2e5cb",
          "yellow300": "#f7da9f",
          "yellow500": "#ffc859",
          "yellow700": "#d4a546",
          "yellow900": "#ab8433",
        },
        "composedSpacings": Object {
          "byte": "10px 15px",
        },
        "fontFamily": Object {
          "color": "#fff",
          "default": "\\"Open Sans\\", sans-serif",
        },
        "fontSize": Object {
          "bit": "11px",
          "byte": "12px",
          "giga": "20px",
          "kilo": "16px",
          "mega": "18px",
        },
        "fontWeight": Object {
          "bold": 700,
          "regular": 400,
        },
        "grid": Object {
          "outerGutter": "8px",
        },
        "spacings": Object {
          "bit": "1px",
          "byte": "5px",
          "giga": "20px",
          "kilo": "10px",
          "mega": "15px",
        },
      }
    }
    withPad={true}
  >
    <Cell__Cell
      width={12}
    >
      Corinthians!
    </Cell__Cell>
  </Grid__Grid>
</styled.section>
`;

exports[`<Grid /> should render the component with the default props 1`] = `
<styled.section
  withPad={false}
>
  <Grid__Grid
    columns={12}
    theme={
      Object {
        "borderRadius": Object {
          "giga": "5px",
          "kilo": "1px",
          "mega": "4px",
        },
        "borderWidth": Object {
          "kilo": "1px",
          "mega": "2px",
        },
        "colors": Object {
          "black": "#000",
          "bodyBackground": "#292d33",
          "danger": "#a94442",
          "green100": "#c6e5cb",
          "green300": "#97e2a3",
          "green500": "#62de76",
          "green700": "#49b85b",
          "green900": "#2f903e",
          "grey100": "#96a2b0",
          "grey200": "#3f454d",
          "grey300": "#343a42",
          "grey400": "#292d33",
          "grey500": "#40464f",
          "mainBackground": "343941",
          "primary100": "#3f9bfd",
          "primary200": "#375f8a",
          "primary300": "#324e70",
          "primary400": "#96a2b0",
          "primary500": "#484e58",
          "red100": "#ffe6e1",
          "red300": "#ffaf9f",
          "red500": "#ff7559",
          "red700": "#d55a41",
          "red900": "#a94442",
          "success": "#49b85b",
          "textColor": "#fff",
          "warning": "#d4a546",
          "white": "#fff",
          "yellow100": "#f2e5cb",
          "yellow300": "#f7da9f",
          "yellow500": "#ffc859",
          "yellow700": "#d4a546",
          "yellow900": "#ab8433",
        },
        "composedSpacings": Object {
          "byte": "10px 15px",
        },
        "fontFamily": Object {
          "color": "#fff",
          "default": "\\"Open Sans\\", sans-serif",
        },
        "fontSize": Object {
          "bit": "11px",
          "byte": "12px",
          "giga": "20px",
          "kilo": "16px",
          "mega": "18px",
        },
        "fontWeight": Object {
          "bold": 700,
          "regular": 400,
        },
        "grid": Object {
          "outerGutter": "8px",
        },
        "spacings": Object {
          "bit": "1px",
          "byte": "5px",
          "giga": "20px",
          "kilo": "10px",
          "mega": "15px",
        },
      }
    }
    withPad={false}
  >
    <Cell__Cell
      width={12}
    >
      Corinthians!
    </Cell__Cell>
  </Grid__Grid>
</styled.section>
`;
